test/extended/testdata/bindata.go

```diff
@@ -6080,6 +6081,57 @@ func testExtendedTestdataDeploymentsGenerationTestYaml() (*asset, error) {
 	return a, nil
 }
 
+var _testExtendedTestdataDeploymentsHookInheritYaml = []byte(`apiVersion: apps.openshift.io/v1
+kind: DeploymentConfig
+metadata:
+  name: hook
+spec:
+  replicas: 1
+  selector:
+    name: hook
+  strategy:
+    type: Rolling
+    rollingParams:
+      pre:
+        failurePolicy: Retry
+        execNewPod:
+          containerName: myapp
+          volumes:
+          - noexist
```

Comment:
I assume this is the key part of this test. (It is always helpful to write a PR description stating why we need this change.)

Have you seen the nonexistent volume not working as expected somewhere? I guess not by just adding the test and e2e passing.

Could we just reuse an existing hook test and add nonexisting volume there?

reference for ignoring non-existing volumes: https://github.com/openshift/origin/blob/714d672734dee9587bdadcde7142aefbb89ba10c/pkg/apps/apis/apps/types.go#L236-L239

Reply:
Thanks for review, will update my PR.